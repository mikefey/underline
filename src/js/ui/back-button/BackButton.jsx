import React from 'react';
import { Link } from 'react-router';
import style from './BackButton.scss';


/**
 * Back button component
 */
const BackButton = (props) => {
  return (
    <Link to={props.link} className={style.wrapper}>
      <div className={style.icon}>
        <div className={style.iconLineOne} />
        <div className={style.iconLineTwo} />
        <div className={style.iconLineThree} />
      </div>
    </Link>
  );
};


/**
 * PropTypes
 * @prop {String} link - The url to navigate to when the button is clicked
 */
BackButton.propTypes = {
  link: React.PropTypes.string.isRequired,
};

export default BackButton;
