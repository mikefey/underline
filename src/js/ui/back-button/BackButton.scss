@import './../../../styles/colors';

.wrapper {
  height: 40px;
  left: 10px;
  position: absolute;
  top: 10px;
  width: 40px;
}

.icon {
  height: 20px;
  left: 5px;
  position: absolute;
  top: 10px;
  width: 30px;
}

.iconLineOne {
  background-color: $dark-grey;
  height: 2px;
  left: 5px;
  position: absolute;
  top: 9px;
  width: 20px;
}

.iconLineTwo {
  background-color: $dark-grey;
  height: 2px;
  left: 3px;
  position: absolute;
  top: 6px;
  transform: rotate(-45deg);
  width: 10px;
}

.iconLineThree {
  background-color: $dark-grey;
  height: 2px;
  left: 3px;
  position: absolute;
  top: 12px;
  transform: rotate(45deg);
  width: 10px;
}
