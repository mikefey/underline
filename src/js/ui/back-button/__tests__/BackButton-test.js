import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import BackButton from './../BackButton.jsx';

test('BackButton component', (t) => {
  const tree = renderToString(<BackButton link={'/'} />);
  t.snapshot(tree);
});
