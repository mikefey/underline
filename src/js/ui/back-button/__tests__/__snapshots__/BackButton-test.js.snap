// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component 1`] = `"<a class=\\"BackButton__wrapper___3allv248BLEnfXA\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"825919163\\"><div class=\\"BackButton__icon___302Gb-yIGqd5oDJ\\" data-reactid=\\"2\\"><div class=\\"BackButton__iconLineOne___3FLnuJCdoUNaVak\\" data-reactid=\\"3\\"></div><div class=\\"BackButton__iconLineTwo___24Pm7a3li3bBJOC\\" data-reactid=\\"4\\"></div><div class=\\"BackButton__iconLineThree___1ES8_m0XRvJqRDJ\\" data-reactid=\\"5\\"></div></div></a>"`;
